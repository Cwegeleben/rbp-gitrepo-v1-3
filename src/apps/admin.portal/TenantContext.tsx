/*
<!-- BEGIN RBP GENERATED: tenant-admin-harden -->
*/
import React, { createContext, useEffect, useState } from "react";
import { getAccessCtx } from "./accessCtx.server";

export const TenantContext = createContext<any>(null);

export const TenantProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [ctx, setCtx] = useState<any>(null);
  useEffect(() => {
    (async () => {
      const c = await getAccessCtx();
      setCtx(c);
    })();
  }, []);
  return <TenantContext.Provider value={ctx}>{children}</TenantContext.Provider>;
};
/*
<!-- END RBP GENERATED: tenant-admin-harden -->
*/
