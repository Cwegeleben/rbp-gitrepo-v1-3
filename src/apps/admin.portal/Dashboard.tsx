/*
<!-- BEGIN RBP GENERATED: tenant-admin-shell -->
*/
import React, { useContext } from "react";
import { TenantContext } from "./TenantContext";
import { KpiTile } from "./KpiTile";
import { EmptyState } from "./EmptyState";
/* <!-- BEGIN RBP GENERATED: tenant-admin-ui-visibility --> */
import { DashboardKPIs } from './app/components/DashboardKPIs';
/* <!-- END RBP GENERATED: tenant-admin-ui-visibility --> */
/* <!-- BEGIN RBP GENERATED: tenant-admin-dashboard-v1 --> */
import { TenantBadge } from './app/components/TenantBadge';
import { PackagerDryRunPanel } from './app/components/PackagerDryRunPanel';
/* <!-- END RBP GENERATED: tenant-admin-dashboard-v1 --> */
/* <!-- BEGIN RBP GENERATED: dev-debug-panel-v1 --> */
import { DevDebugPanel } from './app/components/DevDebugPanel';
/* <!-- END RBP GENERATED: dev-debug-panel-v1 --> */

export const Dashboard: React.FC = () => {
  const ctx = useContext(TenantContext);
  if (!ctx) return <EmptyState message="Loading Tenant Context..." />;
  return (
    <div className="dashboard">
      {/* <!-- BEGIN RBP GENERATED: tenant-admin-dashboard-v1 --> */}
      <div style={{ display: 'flex', alignItems: 'center', justifyContent: 'space-between', marginBottom: 12 }}>
        <TenantBadge />
      </div>
      {/* <!-- END RBP GENERATED: tenant-admin-dashboard-v1 --> */}
      {/* <!-- BEGIN RBP GENERATED: tenant-admin-ui-visibility --> */}
      <div className="kpi-row" style={{ marginBottom: 12 }}>
        <DashboardKPIs />
      </div>
      {/* <!-- END RBP GENERATED: tenant-admin-ui-visibility --> */}
      {/* <!-- BEGIN RBP GENERATED: tenant-admin-dashboard-v1 --> */}
      <div>
        <PackagerDryRunPanel />
      </div>
      {/* <!-- END RBP GENERATED: tenant-admin-dashboard-v1 --> */}
      {/* <!-- BEGIN RBP GENERATED: dev-debug-panel-v1 --> */}
      {ctx?.flags?.showDevTools ? (
        <div style={{ marginTop: 16 }}>
          <DevDebugPanel />
        </div>
      ) : null}
      {/* <!-- END RBP GENERATED: dev-debug-panel-v1 --> */}
    </div>
  );
};
/*
<!-- END RBP GENERATED: tenant-admin-shell -->
*/
