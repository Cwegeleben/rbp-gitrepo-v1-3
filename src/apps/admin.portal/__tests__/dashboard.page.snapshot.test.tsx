/*
<!-- BEGIN RBP GENERATED: tenant-admin-dashboard-v1 -->
*/
import React from 'react';
import { render } from '@testing-library/react';
import { Dashboard } from '../Dashboard';
import { TenantContext } from '../TenantContext';

describe('Dashboard snapshot', () => {
  beforeEach(() => {
    (global as any).fetch = (input: RequestInfo) => {
      const url = typeof input === 'string' ? input : (input as Request).url;
      if (url.includes('/apps/proxy/api/builds')) return Promise.resolve(new Response(JSON.stringify({ pageInfo: { total: 2 } }), { status: 200 }));
      if (url.includes('/apps/proxy/api/catalog/products')) return Promise.resolve(new Response(JSON.stringify({ pageInfo: { total: 3 } }), { status: 200 }));
      if (url.includes('/apps/proxy/api/checkout/package')) return Promise.resolve(new Response(JSON.stringify({ ok: true, meta: { totals: { subtotal: 1, estTax: 0.08, total: 1.08, currency: 'USD' } }, hints: [] }), { status: 200 }));
      return Promise.resolve(new Response('', { status: 404 }));
    };
  });

  it('renders stable UI', () => {
    const ctx = { shopDomain: 'snap.myshopify.com', plan: 'Pro', flags: { showDevTools: true } };
    const { container } = render(
      <TenantContext.Provider value={ctx}>
        <Dashboard />
      </TenantContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
/*
<!-- END RBP GENERATED: tenant-admin-dashboard-v1 -->
*/
