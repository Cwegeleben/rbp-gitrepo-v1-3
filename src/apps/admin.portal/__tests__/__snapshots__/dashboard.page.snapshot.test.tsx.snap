// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Dashboard snapshot renders stable UI 1`] = `
<div>
  <div
    class="dashboard"
  >
    <div
      style="display: flex; align-items: center; justify-content: space-between; margin-bottom: 12px;"
    >
      <div
        style="display: flex; align-items: center; gap: 8px;"
      >
        <div
          style="display: inline-flex; align-items: center; gap: 6px;"
        >
          <span
            aria-label="Tenant snap.myshopify.com"
            style="padding: 4px 8px; border-radius: 999px; background: rgb(238, 242, 255); color: rgb(30, 64, 175); font-weight: 600; font-size: 12px;"
          >
            snap.myshopify.com
          </span>
          <span
            aria-label="Plan Pro"
            style="padding: 2px 6px; border-radius: 6px; background: rgb(241, 245, 249); color: rgb(51, 65, 85); font-size: 11px; border: 1px solid rgb(226, 232, 240);"
          >
            Pro
          </span>
          <span
            style="padding: 2px 6px; border-radius: 6px; background: rgb(255, 247, 237); color: rgb(154, 52, 18); font-size: 11px; border: 1px solid rgb(254, 215, 170);"
            title="Developer tools enabled"
          >
            DEV
          </span>
        </div>
      </div>
    </div>
    <div
      class="kpi-row"
      style="margin-bottom: 12px;"
    >
      <div
        style="display: grid; grid-template-columns: repeat(auto-fit, minmax(180px, 1fr)); gap: 12px;"
      >
        <div
          class="kpi-tile"
          data-tone="default"
          style="border: 1px solid rgb(229, 229, 229); border-radius: 8px; padding: 12px; background: rgb(255, 255, 255);"
        >
          <div
            class="kpi-label"
            style="font-size: 12px; color: rgb(85, 85, 85);"
          >
            Builds
          </div>
          <div
            class="kpi-value"
            style="font-size: 22px; font-weight: 700; color: rgb(17, 17, 17);"
          >
            —
          </div>
        </div>
        <div
          class="kpi-tile"
          data-tone="default"
          style="border: 1px solid rgb(229, 229, 229); border-radius: 8px; padding: 12px; background: rgb(255, 255, 255);"
        >
          <div
            class="kpi-label"
            style="font-size: 12px; color: rgb(85, 85, 85);"
          >
            Catalog
          </div>
          <div
            class="kpi-value"
            style="font-size: 22px; font-weight: 700; color: rgb(17, 17, 17);"
          >
            —
          </div>
        </div>
        <div
          class="kpi-tile"
          data-tone="red"
          style="border: 1px solid rgb(229, 229, 229); border-radius: 8px; padding: 12px; background: rgb(255, 255, 255);"
        >
          <div
            class="kpi-label"
            style="font-size: 12px; color: rgb(85, 85, 85);"
          >
            Package
          </div>
          <div
            class="kpi-value"
            style="font-size: 22px; font-weight: 700; color: rgb(176, 0, 32);"
          >
            Error
          </div>
          <div
            class="kpi-subtext"
            style="font-size: 11px; color: rgb(102, 102, 102); margin-top: 2px;"
          >
            Error
          </div>
        </div>
      </div>
    </div>
    <div>
      <section
        aria-labelledby="packager-dry-run-title"
        style="border: 1px solid rgb(229, 231, 235); border-radius: 8px; padding: 12px;"
      >
        <h3
          id="packager-dry-run-title"
          style="margin: 0px 0px 8px 0px;"
        >
          Dry-run packager
        </h3>
        <div
          aria-live="polite"
          role="status"
          style="color: rgb(107, 114, 128);"
        >
          Running…
        </div>
      </section>
    </div>
    <div
      style="margin-top: 16px;"
    >
      <div
        class="rounded-2xl border border-amber-200 bg-amber-50"
      >
        <button
          aria-controls="dev-debug-content"
          aria-expanded="false"
          class="w-full flex items-center justify-between px-3 py-2"
          type="button"
        >
          <span
            class="text-sm font-semibold text-amber-900"
          >
            Developer Debug
          </span>
          <span
            aria-hidden="true"
            class="text-amber-900"
          >
            ▸
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
